api-gateway/src/main/kotlin/org/cqfn/save/gateway/security/WebSecurityConfig.kt

```diff
@@ -108,11 +135,34 @@ class WebSecurityConfig(
                 RedirectServerAuthenticationFailureHandler("/error")
             )
         }
+        .httpBasic { httpBasicSpec ->
+            // Authenticate by comparing received basic credentials with existing one from DB
+            httpBasicSpec.authenticationManager(
+                UserDetailsRepositoryReactiveAuthenticationManager(
+                    object : ReactiveUserDetailsService {
+                        override fun findByUsername(username: String): Mono<UserDetails> {
```

Comment:
It can, but I leave the `overriding` style intentionally

Most of logic hidden by spring security and such trick with interfaces make it more implicit, as for me
In this case you will not be able to see which method you are override, and looks like you just provide some logic into lambda, which will do something under hood

Reply:
OK, though I think IDEA will highlight it properly and provide a shortcut to a base interface